app/src/models.py

```diff
@@ -48,4 +48,82 @@ def set_password(self, password):
     
     # Check if entered password is correct
     def check_password(self, password):
-        return bcrypt.check_password_hash(self.hashed_password, password)
\ No newline at end of file
+        return bcrypt.check_password_hash(self.hashed_password, password)
+
+# Points Progress
+class Points(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    # establish relationship between 'points' and 'user' model, indicates the points are associated w/ a specific user 
+    user_id = db.Column(db.Integer, ForeignKey('user.id'), nullable=False)
+    points = db.Column(db.Integer, default=0)
+
+# Quiz vs. Task
+class ActivityType(Enum):
+    TASK='Task'
+    QUIZ='Quiz'
+
+class Subject(Enum):
+    CALCULUS='Calculus'
+    LINALG='Linear Algebra'
+    BIOLOGY='Biology'
+    CHEMISTRY='Chemistry'
+    PHYSICS='Physics'
+    ENGLISH='English'
+    FRENCH='French'
+    COMSCI='Computer Science'
+    # ... add/take away as needed
+
+# Quizzes/Tasks
+class Activity(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String(100), nullable=False)
+    # establish relationship between 'quiz' and 'user' model
+    user_id = db.Column(db.Integer, ForeignKey('user.id'), nullable=False)
+    activity_type = db.Column(SQLAlchemyEnum(ActivityType), nullable=False, default=ActivityType.TASK)
+    subject_type = db.Column(SQLAlchemyEnum(Subject), nullable=False, default=Subject.BIOLOGY)
+    start_time = db.Column(db.DateTime, nullable=True, default=func.now()) 
+    end_time = db.Column(db.DateTime, nullable=True, default=func.now())
+    score = db.Column(db.Float, nullable=False)
+    content = db.Column(db.Text, nullable=False) # store task/quiz data
```

Comment:
Probably a bit of a silly question, but what kind of activity content/data were you thinking of storing in this column, given how you're storing questions and answers through a different model? Is this for other general things, e.g. maybe a text description block describing the details of the activity for the users?